### DISCUSSION TITLE
MockServer Expectation Initializers

### DISCUSSION THREAD

**Discussion by PePirex** [testcontainers/testcontainers-java#7753] [Q&A] [answered]
Hi, I searched and couldn't find anything about Expectation Initializers.

Is it possible to pass some parameter, path to file, something like that to the container to initialise the mock server with "default" expectations?
Similar to how one would pass an initializationClass when using the maven plug in.

**Comment by eddumelendez** [comment, marked as answer]
Hi, according to [MockServer docs](https://www.mock-server.com/mock_server/initializing_expectations.html) you can use the env vars

```
MOCKSERVER_WATCH_INITIALIZATION_JSON=true
MOCKSERVER_INITIALIZATION_JSON_PATH=mockserverInitialization.json
```

So, you can do something like

```
var mockServer = new MockServerContainer("image")
    .withEnv("MOCKSERVER_WATCH_INITIALIZATION_JSON", "true")
    .withEnv("MOCKSERVER_INITIALIZATION_JSON_PATH", "mockserverInitialization.json")
    .withCopyFileToContainer(Mountable.fromClasspath("file-in-src-test-resources.json", "mockserverInitialization.json"));
```

**Reply by PePirex** [reply]
This was exactly what I was thinking about. 

Thanks for the help!
